symengine/polynomial.cpp

```diff
@@ -413,11 +413,11 @@ RCP<const UnivariatePolynomial> UnivariatePolynomial::from_vec(const RCP<const S
     return make_rcp<const UnivariatePolynomial>(var, degree, std::move(dict));
 }
 
-RCP<const Basic> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
+RCP<const UnivariatePolynomial> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
 {
     auto iter = d.begin();
     while (iter != d.end()) {
-        if (Expression(0) == iter->second) {
+        if (Expression(0) == iter->second and Expression(0) != iter->first) {
```

Comment:
@CodeMaxx  Can you do one thing and change `UnivariatePolynomial::is_zero()` to return just `dict_.empty()`? Maybe that'll help.